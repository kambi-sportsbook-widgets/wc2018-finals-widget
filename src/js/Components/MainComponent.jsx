import React from 'react'
import PropTypes from 'prop-types'
import { BlendedBackground } from 'kambi-widget-components'
import EventWidget from './EventWidget'
import Carousel from './Carousel'

class MainComponent extends React.Component {
  render() {
    const Events = this.props.data
      .filter(event => event.betOffers.length > 1)
      .map(event => {
        return (
          <EventWidget
            key={event.event.id}
            event={event}
            flagBaseUrl={this.props.flagBaseUrl}
            iconUrl={this.props.iconUrl}
          />
        )
      })

    return (
      <div style={{ padding: '0 .5rem' }}>
        {/*
         * @property backgroundUrl {String} provides path to backgroundImage
         * @property blendWithOperatorColor {Boolean} determines if background should be blended with operator color. (Normally not wanted if providing own background image)
        */}
        <BlendedBackground
          backgroundUrl={this.props.backgroundUrl}
          blendWithOperatorColor={this.props.blendWithOperatorColor}
        />
        <Carousel slides={Events} />
      </div>
    )
  }
}

MainComponent.propTypes = {
  data: PropTypes.array.isRequired,
  backgroundUrl: PropTypes.string.isRequired,
  blendWithOperatorColor: PropTypes.bool.isRequired,
  flagBaseUrl: PropTypes.string,
  iconUrl: PropTypes.string,
}

export default MainComponent
