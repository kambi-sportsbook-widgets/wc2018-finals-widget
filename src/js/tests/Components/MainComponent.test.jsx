// Don't load svg strings into tests
jest.mock('../../../../node_modules/kambi-widget-components/src/Carousel/chevron_left.svg', () => <svg>Placeholder</svg>)
jest.mock('../../../../node_modules/kambi-widget-components/src/Carousel/chevron_right.svg', () => <svg>Placeholder</svg>)

import React from 'react'
import ReactShallowRenderer from 'react-test-renderer/shallow'
import MainComponent from '../../Components/MainComponent'

let renderer
const emptyData = []
const exampleData = [
  { event: { id: 1 } },
  { event: { id: 2 } },
  { event: { id: 3 } },
]

describe('MainComponent DOM rendering', () => {

  beforeEach(() => {
    renderer = new ReactShallowRenderer()
  })

  it('fails gracefully with empty data array', () => {
    expect(
      renderer.render(
        <MainComponent
          data={emptyData}
          backgroundUrl={''}
          blendWithOperatorColor={true}
        />
      )
    )
    .toMatchSnapshot()
  })

  it('renders correctly with example data', () => {
    expect(
      renderer.render(
        <MainComponent
          data={exampleData}
          backgroundUrl={''}
          blendWithOperatorColor={true}
        />
      )
    )
    .toMatchSnapshot()
  })
})
