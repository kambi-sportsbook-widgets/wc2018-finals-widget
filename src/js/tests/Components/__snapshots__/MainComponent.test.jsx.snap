// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainComponent DOM rendering fails gracefully with empty data array 1`] = `
<div>
  <BlendedBackground
    backgroundUrl=""
    blendWithOperatorColor={true}
  />
  <Carousel
    animationType="slide"
    autoPlay={false}
    cssEase="ease"
    height="100%"
    imageLoaded={[Function]}
    indicatorLiStyles={Object {}}
    indicatorUlStyles={Object {}}
    infinite={true}
    infiniteLoop={true}
    initializedCarousel={[Function]}
    intervalDuration={3500}
    legendClassName={null}
    onCarouselChange={[Function]}
    onCarouselItemClick={[Function]}
    onCarouselMouseEnter={[Function]}
    onCarouselMouseLeave={[Function]}
    redirectCallback={null}
    selectedItem={0}
    showArrows={true}
    showIndicators={false}
    stopOnHover={false}
    transitionDuration={800}
    wrapperClassName={null}
  />
</div>
`;

exports[`MainComponent DOM rendering renders correctly with example data 1`] = `
<div>
  <BlendedBackground
    backgroundUrl=""
    blendWithOperatorColor={true}
  />
  <Carousel
    animationType="slide"
    autoPlay={false}
    cssEase="ease"
    height="100%"
    imageLoaded={[Function]}
    indicatorLiStyles={Object {}}
    indicatorUlStyles={Object {}}
    infinite={true}
    infiniteLoop={true}
    initializedCarousel={[Function]}
    intervalDuration={3500}
    legendClassName={null}
    onCarouselChange={[Function]}
    onCarouselItemClick={[Function]}
    onCarouselMouseEnter={[Function]}
    onCarouselMouseLeave={[Function]}
    redirectCallback={null}
    selectedItem={0}
    showArrows={true}
    showIndicators={false}
    stopOnHover={false}
    transitionDuration={800}
    wrapperClassName={null}
  >
    <EventWidget
      event={
        Object {
          "event": Object {
            "id": 1,
          },
        }
      }
      flagBaseUrl={undefined}
      iconUrl={undefined}
    />
    <EventWidget
      event={
        Object {
          "event": Object {
            "id": 2,
          },
        }
      }
      flagBaseUrl={undefined}
      iconUrl={undefined}
    />
    <EventWidget
      event={
        Object {
          "event": Object {
            "id": 3,
          },
        }
      }
      flagBaseUrl={undefined}
      iconUrl={undefined}
    />
  </Carousel>
</div>
`;
